rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/package.scala

```diff
@@ -61,45 +71,51 @@ package object matcher {
     def emptyMap[A]: OptionalFreeMapWithCost[A] =
       StateT((m: FreeMap) => {
         OptionT(
-          State((c: CostAccount) => {
-            (c, None)
+          StateT((c: CostAccount) => {
+            Right((c, None))
           })
         )
       })
 
     def pure[A](value: A): OptionalFreeMapWithCost[A] =
       StateT((m: FreeMap) => {
         OptionT(
-          State((c: CostAccount) => {
-            (c, Some((m, value)))
+          StateT((c: CostAccount) => {
+            Right((c, Some((m, value))))
           })
         )
       })
 
     def liftF[A](option: Option[A]): OptionalFreeMapWithCost[A] =
       StateT((m: FreeMap) => {
-        OptionT(State((c: CostAccount) => {
-          (c, option.map(m -> _))
+        OptionT(StateT((c: CostAccount) => {
+          Right((c, option.map(m -> _)))
         }))
       })
   }
 
   type NonDetFreeMapWithCost[A] = StateT[StreamWithCost, FreeMap, A]
-  type StreamWithCost[A]        = StreamT[State[CostAccount, ?], A]
+  type StreamWithCost[A]        = StreamT[StateT[Either[OutOfPhlogistonsError.type, ?], CostAccount, ?], A]
```

Comment:
Please extract additional type aliases as needed so that no type alias uses more than one monad transformer